### PR TITLE
Feature decorate a ruby class

### PR DESCRIPTION
**Pull request by Beardless-sheik**
In this pull request: 
- Crate Nameable Class
- Create Base Decorator class that inherits from Nameable class
- Create 2 Decorators: CapitalizeDecorator & TrimmerDecorator that implent the Base Decorator class methods to extend the "correct_name" functionality method

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Capitalise Decorator: A decorator that implements the correct_name method in the Nameable class differently
+class CapitalizeDecorator < Base_Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (Cathella) [inline nameable.rb]: - [ ] [REQUIRED] To Capitalize the first letter of the given name, I suggest that you use the `.capitalize` method instead. LIne 27 will instead become `@nameable.correct_name.capitalize`

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  end
+end
+
+# Trimmer Decorator: A decorator that implements the correct_name method in the Nameable class differently
+class TrimmerDecorator < Base_Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
+  end
```
Comment (Cathella) [inline nameable.rb:35]: Great use of the slice method 👍🏻!

You could also compare the length of the string in question like this `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @Beardless-sheik 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct Github Flow.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

i-guess-we-have-to-give-it-another-shot-liz-lemon


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Beardless-sheik 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**